src/scraper/http.cpp

```diff
@@ -138,6 +138,49 @@ namespace
     };
 
 #endif
+
+    //!
+    //! \brief Parse an etag value from an HTTP header field.
+    //!
+    //! This will parse etag values from headers in these standard formats:
+    //!
+    //!   ETag: "12345"
+    //!   ETag: W/"12345"
+    //!
+    //! The name of the header is matched in a case-insensitive fashion. This
+    //! function will return an empty string for non-standard, malformed, and
+    //! non-etag headers. It removes the quotes from the output and ignores a
+    //! space after the colon that separates the header name from the value.
+    //!
+    //! \param header Entire HTTP header field that includes the name and value.
+    //!
+    //! \return The parsed etag value or an empty string if the supplied header
+    //! contains no standard etag content.
+    //!
+    std::string ParseEtag(const std::string& header)
+    {
+        if (header.size() <= 8 || header[4] != ':') {
+            return std::string();
+        }
+
+        constexpr char expected[] = "etag";
+        constexpr int32_t to_upper = 32;
+
+        for (size_t i = 0; i < 4; ++i) {
+            if (header[i] != expected[i] && header[i] != expected[i] - to_upper) {
+                return std::string();
+            }
+        }
+
+        const size_t start_quote = header.find('"', 5);
+        const size_t end_quote = header.find('"', start_quote + 1);
+
+        if (start_quote == std::string::npos || end_quote == std::string::npos) {
+            return std::string();
+        }
```

Comment:
The standard requires quotes, but are we guaranteed that? The enhanced split function I wrote works whether there are quotes or not...

Reply:
If a web server is well-behaved, it will include both quotes around the value of the etag since it is common for header values to begin with a space character after the colon separator, and the etag value can potentially contain a space as well. The quotes disambiguate the (optional) leading space from the value of the etag.

Reply:
If the server is not well-behaved, I would be concerned about whether we're communicating with the project server to begin with. I decided to follow the spec for this one since I'm not sure that we actually want to continue to process a non-standard response&mdash;it could be a signal that something else is wrong.

Reply:
Ref: https://tools.ietf.org/html/rfc2616#section-4.2

Reply:
This is immune to the extra CR problem that we were having before, just like my version.

Reply:
Ok.